opamp/observiq/reload_funcs.go

```diff
@@ -0,0 +1,216 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observiq
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/observiq/observiq-otel-collector/opamp"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+func managerReload(client *Client, managerConfigPath string) opamp.ReloadFunc {
+	return func(contents []byte) (bool, error) {
+		// Unmarshal config and only pull fields out that are allowed to be updated.
+		var newConfig opamp.Config
+		if err := yaml.Unmarshal(contents, &newConfig); err != nil {
+			return false, fmt.Errorf("failed to validate config %s", ManagerConfigName)
+		}
+
+		// Read in existing config file
+		currConfig, err := opamp.ParseConfig(managerConfigPath)
+		if err != nil {
+			return false, err
+		}
+
+		// Check if the updatable fields are equal
+		// If so then exit
+		if currConfig.CmpUpdatableFields(newConfig) {
```

Comment:
@cpheps should we be comparing to what is currently in the manager.yml file or what is currently stored in the agent config manager?

what if the user had changed the manager.yml and set some new values. this does nothing right? then they change those same values in iris to be sent down. now when this comparison happens, they are equal and no changes actually occur right?

Reply:
Yes this opens up an issue where if the on-disk copy matches iris but the in memory doesn't it'll never be updated. I'll look into adding in memory protection.